operators/src/source/gdal_source.rs

```diff
@@ -372,6 +374,155 @@ impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
     }
 }
 
+#[derive(PartialEq, Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct GdalMetadataFixedTimes {
+    pub time_steps: Vec<TimeInterval>,
+    pub params: GdalDatasetParameters,
+    pub result_descriptor: RasterResultDescriptor,
+    pub time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+}
+
+impl GdalMetadataFixedTimes {
+    #[allow(dead_code)]
+    fn new(
+        time_steps: Vec<TimeInterval>,
+        params: GdalDatasetParameters,
+        result_descriptor: RasterResultDescriptor,
+        time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+    ) -> Self {
+        let mut sorted_time_steps = time_steps;
+        sorted_time_steps.sort_by(|a, b| a.end().partial_cmp(&b.start()).unwrap());
+
+        GdalMetadataFixedTimes {
+            time_steps: sorted_time_steps,
+            params,
+            result_descriptor,
+            time_placeholders,
+        }
+    }
+}
+
+#[async_trait]
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for GdalMetadataFixedTimes
+{
+    async fn loading_info(&self, query: RasterQueryRectangle) -> Result<GdalLoadingInfo> {
+        let minimum_step = TimeStep {
+            granularity: TimeGranularity::Millis,
+            step: 1,
+        };
+
+        let sorted_time_steps = if self
```

Comment:
you already sort them in the `new` method why sort again?

Reply:
Fixed, moved all relevant code to the `new` method